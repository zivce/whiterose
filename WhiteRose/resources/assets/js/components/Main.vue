<template>
    <div id="main">
      <transition name="fade">
        <matrix v-if="isVisibleMatrix"></matrix>
      </transition>
      <transition name="fade">
        <router-view v-if="!isVisibleMatrix"></router-view>
      </transition>
    </div>
</template>


<script>
import logger from "../utils/groupLogger";
import Matrix from "./MatrixLoader.vue";

export default {
  components: {
    Matrix
  },
  mounted() {
    logger(["Component Main mounted", "happy hacking"], "Main.vue");

    this.timer = setTimeout(() => {
      this.isVisibleMatrix = false;
    }, 4000);
  },
  destroyed() {
    clearTimeout(this.timer);
  },
  data() {
    return {
      timer: undefined,
      isVisibleMatrix: true
    };
  }
};
</script>


<style scoped>
/* da se pomeri na dole kad predje na sledecu komponentu  */
.fade-enter-active,
.fade-leave-active {
  transition: opacity 0.4s;
}
.fade-enter,
.fade-leave-to {
  opacity: 0;
}
</style>
