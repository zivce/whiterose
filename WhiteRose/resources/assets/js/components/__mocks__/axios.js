module.exports = {
    post: jest.fn(() => Promise.resolve({ data: 3 }))
  }