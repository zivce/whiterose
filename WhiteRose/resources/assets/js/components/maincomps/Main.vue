<template>
  <div id="main_vue">
      <div class="row" id="header">
        <h1 >
        <router-link 
        to="/home"
        >
          Home
        </router-link>
      </h1>
        
        <b-dropdown  
        :text="username" id="user"

        >
          <b-dropdown-item :to="profile_url">
            <slot name="button-content"><icon name="user"></icon> Profile</slot>
          </b-dropdown-item>
        
          <b-dropdown-item  @click="handleLogout()">
              <slot name="button-content"><icon name="logout"></icon> Logout</slot>
          </b-dropdown-item>
          
        </b-dropdown>

      </div>
      
      <div id="main_main">

        <home v-if="home">
        </home>
<!--         
        <home v-if="main_comp" id="">
          
        </home> -->


        <transition name="fade" mode="out-in" >
          <router-view v-if="!home">
          </router-view>
        </transition>
      </div>
        <div id="footer" class="row">
          <p class="d-flex">You've reached rock bottom. Who lives in the pineapple under the sea...</p>
      </div>
      
      <vue-snotify/>

      
  </div>

</template>

<script>
import Home from "./Home.vue";
import Icon from "vue-awesome/components/Icon";
import "vue-awesome/icons/user";
import "../../Icons/logout";
import eventBus from "../../utils/eventBus";

export default {
  components: {
    Home,
    Icon
  },
  mounted() {
    this.username = this.$store.getters.returnEmail;

    this.user_id = this.$store.getters.returnId;

    if (this.$store.state.route.path === "/") {
      this.main_comp = true;
    } else {
      this.main_comp = false;
    }

    this.profile_url = `/user/${this.user_id}/dashboard/setup`;
  },
  computed: {
    home() {}
  },
  methods: {
    handleLogout() {
      axios.get("/clientlogout").then(res => {
        if (res.status === 200) {
          this.notifySuccess("Logged out!", "Success!");

          window.setTimeout(() => {
            window.location.reload();
          }, 1500);
        } else {
          this.errorToast("Error happened", "Error!");
        }
      });
    }
  },
  data() {
    return {
      user_id: "",
      profile_url: ``,
      username: "",
      main_comp: false
    };
  }
};
</script>

<style scoped>
#main_main {
  min-height: 90vh;
}

#header >>> #user__BV_toggle_ {
  margin: auto;
  height: 5vh;
  background-color: transparent;
  border: 0;
}
#header >>> .dropdown-menu.show {
  top: 4px !important;
  left: -70px !important;
}
#header >>> #user {
  position: absolute;
  right: 5vw;
  top: 4vh;
}
#header >>> #user__BV_toggle_:focus {
  outline: none;
  border: none;
  box-shadow: none;
}
#header h1 {
  height: 5vh;
  margin: 0 auto;
}

#footer p {
  margin: auto;
}
#footer {
  text-align: center;
  margin-top: 5%;
  box-shadow: 3px 3px 7px 2px #e2e2e2;
  height: 13vh;
  color: white;
  background: rgba(0, 0, 53, 0.93);
}
#header {
  margin-bottom: 3%;
  box-shadow: 3px 3px 7px 2px #e2e2e2;
  height: 13vh;
  color: white;
  background: rgba(0, 0, 53, 0.93);
}

#main_vue {
  min-height: 100vh;
  /* background: #C33764;  /* fallback for old browsers */
  /* background: -webkit-linear-gradient(to right, #1D2671, #C33764);  Chrome 10-25, Safari 5.1-6 */
  /* background: linear-gradient(to right, #1D2671, #C33764); W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
}
</style>
