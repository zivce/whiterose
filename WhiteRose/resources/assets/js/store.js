import Vue from 'vue';

import Vuex from 'vuex';
Vue.use(Vuex);

const store = new Vuex.Store({
    state: {
        user: {},
        role : '',
        email : '',
        id : '',
        sites : '',
        scans: ''
    },
    mutations : {
        setUser(state,payload)
        {
            state.role = payload.role;
            state.user = payload;
            state.email = payload.email;
            state.id = payload.id;
        },
        setSites(state,payload)
        {
            state.sites = payload;
        },
        setScans(state,payload)
        {
            state.scans  = payload;
        }
    },
    getters:{
        returnParams : state => {
            return state.route.params;
        },
        returnRoute : state => {
            return state.route.fullPath;
        },
        returnRole : state =>{
            return state.role;
        },
        returnUser : state => {
            return state.user;
        },
        returnEmail : state =>{
            return state.email;
        },
        returnId : state => {
            return state.id;
        }

    }

})   
export default store;