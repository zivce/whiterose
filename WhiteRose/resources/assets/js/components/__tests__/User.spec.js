
import {shallowMount, mount, createLocalVue} from '@vue/test-utils';
import {createRenderer} from 'vue-server-renderer';

import Main from '@/maincomps/MainChildren/User.vue';

window.Vue = require("vue");
window.Vue = require("vue");
window.moment = require("moment");
import * as jsdom from 'jsdom';
/**
 * Imports block
 */

import BootstrapVue from "bootstrap-vue";
import VueRouter from "vue-router";
import Snotify, { SnotifyPosition } from "vue-snotify";
import Vuex from "vuex";
import VeeValidate from "vee-validate";
import axios from "axios";
import errorToaster from "@/toastr/FormErrorToaster";
import successToastr from "@/toastr/welcometoastr";

import checkFields from "../../utils/checkAllFields";
import fixInputs from '../../mixins/fixNumberInput';


/**
 * Comps
 */


/**
 * Mixins
 */

Vue.mixin(errorToaster);
Vue.mixin(checkFields);
Vue.mixin(successToastr);
Vue.mixin(fixInputs);
/**
 * What Vue should globally use.
 */
Vue.use(Snotify);
Vue.use(BootstrapVue);
Vue.use(VeeValidate);
import Icon from "vue-awesome/components/Icon";
import "vue-awesome/icons/info";
import "vue-awesome/icons/wrench";
import "vue-awesome/icons/comment";
import "vue-awesome/icons/user";

import store from '../../store';

const localVue = createLocalVue();

localVue.component(Icon);
localVue.use(VueRouter);


//THIS MOCKS THE USER VUE 

const routes = [
    {
        name:"root",
        path:"/",
        component : require('@/maincomps/MainChildren/User.vue')
    },
    {
        name:"messages",
        path : 'messages',
        component : require('../maincomps/MainChildren/Pentester.vue')
    }
]
const router = new VueRouter({
    routes
})
localVue.config.ignoredElements = ['router-view']

describe("MainClient.vue", () => {
    
    const wrapper = shallowMount(Main,
        {localVue,router,
        store,
        stubs : {
            'Messages' : '<div class="msgs"/>'
            
        }});
    it("correctly mounts the vue instance", () => {
        expect(wrapper.isVueInstance()).toBeTruthy();
    })
    it("correctly shows the info modal after click of info btn", () => {
        const info = wrapper.find({ref : 'info_btn'})
        info.trigger('click');
        const t = wrapper.find({ref : "user_info"});
        expect(t).toBeTruthy();
    })

    it("correctly renders the list of navigation buttons", () => {
        const btns = wrapper.find('.nav_btns')
    
        expect(btns).toBeTruthy();
    })

    it("correctly renders the list of user navigation", () => {
        const nav_user = wrapper.find('.user_nav_main')
    
        expect(nav_user).toBeTruthy();
    })
    
    it("has same html structure", () => {
        const renderer = createRenderer();
        renderer.renderToString(wrapper.vm, (err,str) => {
            if(err) throw new Error(err)
            expect(str).toMatchSnapshot();
        })
    })

    it("correctly renders the child component", () => {
        const chld = wrapper.findAll('.msgs');
        expect(chld).toBeTruthy();
    })
   

    it("correctly adds class to an active navigation button", () => {
        let btn = wrapper.find('.nav-item');
        btn.trigger('click');
        let anchor = btn.find('.active')
        expect(anchor).toBeTruthy();
    })


    it("correctly communicates with vuex ", () => {
        wrapper.vm.$store.commit("setTest",{test:3});
        expect(wrapper.vm.$store.getters.returnTest).toBe(3);
    })

    it("correctly uses vue router" , () => {
        wrapper.vm.$router.push({name: "messages"});
        const ref = wrapper.find({ref: "pentester"})
        expect(ref).toBeTruthy();

    })
});
