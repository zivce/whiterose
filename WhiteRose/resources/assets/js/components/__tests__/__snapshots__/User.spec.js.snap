// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainClient.vue has same html structure 1`] = `
<div id="user_comps">
  <div class="nav_btns">
    <button type="button" class="btn btn btn-client btn-secondary">
      <!---->
    </button>
    <!---->
    <!---->
    <!---->
  </div>
  <!---->
  <ul class="d-flex justify-content-center user_nav_main nav nav-pills">
    <li class="nav-item nav_item_user">
      <!---->
    </li>
    <li class="nav-item nav_item_user">
      <!---->
    </li>
    <li class="nav-item nav_item_user">
      <!---->
    </li>
    <li class="nav-item nav_item_user">
      <!---->
    </li>
    <li class="nav-item nav_item_user">
      <!---->
    </li>
  </ul>
  <!---->
  <!---->
</div>
`;
